### PR TITLE
LLT-4688: Add logs to UDP `sendto()/recvfrom()` socket calls

### PR DESCRIPTION
**Pull request by matszczygiel**
I've used `std::io::Error::last_os_error()` for extracting the error code and message from syscalls.
The error is then logged out at the call site

### CODE REVIEW

File: boringtun/src/device/mod.rs [hunk 1]
```diff
@@ -1040,23 +1053,29 @@ impl Device {
...
-                                tracing::trace!(
-                                    message = "Writing packet to network v6",
-                                    interface = ?t.iface.name(),
-                                    packet_length = packet.len(),
-                                    src_addr = ?addr,
-                                    public_key = public_key
-                                );
+                                if let Err(err) = udp6.sendto(packet, addr) {
+                                    tracing::warn!(message = "Failed to write packet to network v6", error = ?err);
+                                }else {
```
Comment (tomaszklak) [inline boringtun/src/device/mod.rs:1070]: formatting seems to be broken

Comment (matszczygiel) [reply boringtun/src/device/mod.rs:1070]: Yeah, and I don't know why. It seems rustfmt doesn't work on this part for some reason.
I'll fix it manually

Comment (matszczygiel) [reply boringtun/src/device/mod.rs]: Fixed

**Review by tomaszklak — COMMENTED** [review]
+1
One thing to consider, is that maybe it would be better to also add in the new logs, more information - like to which address are we sending, on which address are we listening? This might not always be available but where possible I think it would be useful.

**Review by dfetti — APPROVED** [review]
+1.0

**Comment by matszczygiel** [comment]
@tomaszklak I've included the endpoint in the logs

**Pull request merged**